paddle/framework/operator.h

```diff
@@ -110,29 +98,32 @@ class OperatorBase {
   std::shared_ptr<std::unordered_map<std::string, int>> in_out_idxs_;
 };
 
-class KernelContext {
+class OperatorContext {
  public:
-  KernelContext(const OperatorBase* op, const std::shared_ptr<Scope>& scope,
-                const platform::DeviceContext& device_context)
-      : op_(*op), scope_(scope), device_context_(device_context) {}
+  OperatorContext(const OperatorBase* op, const std::shared_ptr<Scope>& scope)
+      : op_(*op), scope_(scope) {}
+
+  int InputSize() const { return static_cast<int>(op_.inputs_.size()); }
 
-  const Variable* Input(int index) const {
+  int OutputSize() const { return static_cast<int>(op_.outputs_.size()); }
+
+  const Variable* InputVar(int index) const {
```

Comment:
In `OperatorBase` , `Input` returns `string`, In `Context`, `Input` returns `Tensor`, and here is another `InputVar`

Can we uniform all the `Input()`s, and use a single API like `template <typename T> input(std::string)`, and implements three types: `string`, `Tensor`, `Variable`. 

This is much simplier to understand. @jacquesqiao